ebean-datasource/src/main/java/io/ebean/datasource/pool/PooledConnection.java

```diff
@@ -82,12 +108,14 @@ final class PooledConnection extends ConnectionDelegator {
    */
   private boolean failoverToReadOnly;
   private boolean resetAutoCommit;
-  private boolean resetSchema;
-  private boolean resetCatalog;
-  private boolean initialisedSchema;
-  private boolean initialisedCatalog;
-  private String currentSchema;
-  private String currentCatalog;
+  private int schemaState = SCHEMA_CATALOG_UNKNOWN;
+  private int catalogState = SCHEMA_CATALOG_UNKNOWN;
+
+  // this is used for cache computation
+  private String cacheKeySchema = UNKNOWN;
+  private String cacheKeyCatalog = UNKNOWN;
```

Comment:
I have renamed `currentSchema` to `cacheKeySchema`, because it clarifies, for which purpose this value is used.